__init__.py

```diff
@@ -35,28 +34,94 @@ class SkillInstallerSkill(MycroftSkill):
     def __init__(self):
         super(SkillInstallerSkill, self).__init__(name="SkillInstallerSkill")
 
-    def initialize(self):
-        install = IntentBuilder("InstallIntent"). \
-            require("InstallKeyword").build()
-        self.register_intent(install, self.install)
-
+    @intent_handler(IntentBuilder("InstallIntent").require("Install"))
     def install(self, message):
         utterance = message.data.get('utterance').lower()
-        skill = utterance.replace(message.data.get('InstallKeyword'), '')
+        name = utterance.replace(message.data.get('InstallKeyword'), '')
         self.speak_dialog("installing")
-        text = os.popen(
-            BIN + " install " + skill.strip().replace(" ", "-")).read()
-        if text.splitlines()[1] == "Your search has multiple choices":
-            stdout = text.splitlines()
-            del stdout[0:3]
-            self.speak_dialog("choose", data={'skills': ", ".join(stdout)})
-        elif text.splitlines()[1] == "Skill not found" or skill == "":
-            self.speak_dialog("not.found", data={'skill': skill})
-        elif text.splitlines()[2] == "Skill installed!":
-            self.speak_dialog("installed", data={'skill': skill})
-
-    def stop(self):
-        pass
+
+        # Invoke MSM to perform installation
+        try:
+            output = subprocess.check_output(
+                [BIN, 'install', '"' + skill.strip() + '"'],
```

Comment:
The `"` are superfluous when using the arg list method like this. This will actually result in `"\"WORD1 WORD2\""` which msm will have trouble finding.

Simple illustrative case:

```python
import subprocess

output = subprocess.check_output(['echo', '"Hello"'])
print output
```

Reply:
Yep, stupid conversion of old code on my part